File: src/main/java/jp/skypencil/CloseableValue.java
package jp.skypencil;

import org.objectweb.asm.tree.analysis.Value;

/** A stateless {@link Value} implementation that holds state of {@link AutoCloseable} instance. */
final class CloseableValue implements Value {
  private final boolean closed;

  CloseableValue(boolean closed) {
    this.closed = closed;
  }

  @Override
  public int getSize() {
    return 1;
  }

  boolean isClosed() {
    return this.closed;
  }
}

File: src/main/kotlin/jp/skypencil/CloseableValue.kt
package jp.skypencil

import org.objectweb.asm.tree.analysis.Value

/**
 * A stateless [Value] implementation that holds state of [AutoCloseable] instance.
 */
class CloseableValue(private val closed: Boolean) : Value {
    override fun getSize(): Int = 1

    fun isClosed(): Boolean = closed
}
